options/options.go

```diff
@@ -82,7 +83,13 @@ type TerragruntOptions struct {
 func NewTerragruntOptions(terragruntConfigPath string) *TerragruntOptions {
 	workingDir := filepath.Dir(terragruntConfigPath)
 
-	downloadDir := filepath.Join(os.TempDir(), "terragrunt")
+	user, err := user.Current()
+	if err != nil {
+		fmt.Fprintf(os.Stderr, "error: %v\n", err)
+		os.Exit(1)
+	}
+
+	downloadDir := filepath.Join(user.HomeDir, ".terragrunt")
```

Comment:
Unfortunately, this approach makes cross-compiling harder. In the past, I've used https://github.com/mitchellh/go-homedir.